socorro/lib/transform_rules.py

```diff
@@ -47,12 +49,61 @@ class Rule(RequiredConfig):
         default=False,
     )
 
-
     #--------------------------------------------------------------------------
     def __init__(self, config=None, quit_check_callback=None):
         self.config = config
         self.quit_check_callback = quit_check_callback
 
+    def _send_to_sentry(self, tag, *args, **kwargs):
+        """Execute this when an exception has happened only.
+        If self.config.sentry.dsn is set up, it will try to send it
+        to Sentry. If not configured, nothing happens.
+        """
+        try:
+            dsn = self.config.sentry.dsn
+        except KeyError:
+            # if self.config is not a DotDict, we can't access the sentry.dsn
+            dsn = None
+        if dsn:
+            extra = {
+                'class': self.__class__.__name__,
+                'tag': tag,  # this can 'predicate' or 'action'
+            }
+
+            # For every crash acted on, it's always
+            # act(raw_crash, ...) etc.
+            # But be defensive in case the first argument isn't there,
+            # isn't a dict or doesn't have a 'uuid'.
+            if args and isinstance(args[0], collections.Mapping):
```

Comment:
I'd make two changes.

First, change `_send_to_sentry` to also take the function that kicked up the error.

```
def _send_to_sentry(self, tag, fun, *args, **kwargs):
```

Then change all the uses of `_send_to_sentry` so that they pass in `self._predicate` or whatever function it is.

Then change the above code along these lines:

```
argspec = inspect.getcallargs(fun, *args, **kwargs)
if 'raw_crash' in argspec:
    crash_id = argspec['raw_crash'].get('uuid')
    ...
```

Why do it this way? It's more specific and resilient to changes. It uses existing information in the function signature and how Python binds call values to function arguments.